examples/merkle_proofs.md

```diff
@@ -0,0 +1,157 @@
+This example demonstrates the _Merkle proof_ feature of Irmin on the task of producing tamper-proof banking account statements.
+
+From a high-level perspective, a Merkle proof is a Merkle tree where some subtrees are missing and replaced by their hash. A Merkle tree is simply a DAG where the hash of a node is derived from the hash of its children.
```

Comment:
"From a high-level perspective, a Merkle proof is a path in a Merkle tree where the missing subtrees are replaced by their hash."
Could this be a more strict intuition?

Reply:
Our Merkle proofs can capture more structure than a single path from the root to some leaf: given a complete Merkle tree, one can choose to "prune" each of its subtrees (replacing the underlying structure with a hash) zero or more times, and the result is a Merkle proof.

The terminology is quite conflicting in the wild: some people use "Merkle proof" to talk about only the linear path-like case, and others like us are using the more generalised concept.

Reply:
If in the generalised concept a Merkle proof is the result of pruning a Merkle tree then a more accurate terminology for the result could be "set of paths" or  "Merkle tree skeleton" or something else. The idea is that I'm not sure that the pruning operation renders a Merkle tree. Also, if the more more generalised concept already has a definition in the literature, then let's talk about it.

Reply:
Via https://blog.ethereum.org/2015/11/15/merkling-in-ethereum/ 
"A Merkle proof consists of a chunk, the root hash of the tree, and the “branch” consisting of all of the hashes going up along the path from the chunk to the root. Someone reading the proof can verify that the hashing, at least for that branch, is consistent going all the way up the tree, and therefore that the given chunk actually is at that position in the tree."

This is consistent with:

A **Merkle proof** for a given `key` is the tuple `(key, value, path, root)`, where:

* `value` is the content of the leaf at the position described by `key`,
* `path` is the "sequence of siblings" of the branch described by `key`, and
* `root` is the expected root of the tree.

["chunk"/`value`]